lib/models/task.dart

```diff
@@ -1,57 +1,63 @@
 import 'package:flutter/material.dart';
 import 'package:json_annotation/json_annotation.dart';
-import 'package:vikunja_app/components/date_extension.dart';
 
 import 'package:vikunja_app/models/label.dart';
 import 'package:vikunja_app/models/user.dart';
 import 'package:vikunja_app/models/taskAttachment.dart';
-import 'package:vikunja_app/theme/constants.dart';
 import 'package:vikunja_app/utils/checkboxes_in_text.dart';
 
 @JsonSerializable()
 class Task {
-  final int? id, parentTaskId, priority, listId, bucketId;
-  final DateTime? created, updated, dueDate, startDate, endDate;
-  final List<DateTime?>? reminderDates;
-  final String identifier;
-  final String? title, description;
+  final int id;
+  final int? parentTaskId, priority, bucketId;
+  final int listId;
+  late final DateTime created, updated;
+  final DateTime? dueDate, startDate, endDate;
+  final List<DateTime> reminderDates;
+  final String? identifier;
+  final String title, description;
   final bool done;
-  final Color color;
+  final Color? color;
   final double? kanbanPosition;
-  final User? createdBy;
+  final User createdBy;
   final Duration? repeatAfter;
-  final List<Task>? subtasks;
-  final List<Label>? labels;
-  final List<TaskAttachment>? attachments;
+  final List<Task> subtasks;
+  final List<Label> labels;
+  final List<TaskAttachment> attachments;
   // TODO: add position(?)
 
-  CheckboxStatistics? _checkboxStatistics;
+  late final CheckboxStatistics checkboxStatistics = getCheckboxStatistics(description);
+  late final hasCheckboxes = checkboxStatistics.total != 0;
+  late final textColor = (color != null && color!.computeLuminance() > 0.5) ? Colors.black : Colors.white;
+  late final hasDueDate = dueDate?.year != 1;
 
-  // // TODO: use `late final` once upgraded to current dart version
   Task({
-    required this.id,
-    required this.identifier,
-    this.title,
-    this.description,
+    this.id = -1,
+    this.identifier,
+    this.title = '',
+    this.description = '',
     this.done = false,
-    this.reminderDates,
+    this.reminderDates = const [],
     this.dueDate,
     this.startDate,
     this.endDate,
     this.parentTaskId,
     this.priority,
     this.repeatAfter,
-    this.color = vBlue, // TODO: decide on color
+    this.color,
```

Comment:
Made `color` nullable again (here and for Label). If it's non-nullable then every task that is edited will have the color that is set here have that color set for the task on the server. @Benimautner if you don't like the default look of the task cards in the kanban view that probably should be changed on the UI side, not here.

Reply:
That was something I wanted to fix, because the way it looked after I changed that was horrible. Just haven't gotten around to it. Thanks!